demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -82,60 +84,74 @@
     #define NETWORK_BUFFER_SIZE    ( 1024U )
 #endif
 
+/**
+ * @brief The largest random number to use in client identifier.
+ *
+ * @note Random number is added to MQTT client identifier to avoid client
+ * identifier collisions while connecting to MQTT broker.
+ */
+#define MAX_RAND_NUMBER_FOR_CLIENT_ID           ( 999u )
+
+/**
+ * @brief Maximum number of random number digits in Client Identifier.
+ * @note The value is derived from the #MAX_RAND_NUM_IN_FOR_CLIENT_ID.
+ */
+#define MAX_RAND_NUMBER_DIGITS_FOR_CLIENT_ID    ( 3u )
+
 /**
  * @brief Length of MQTT server host name.
  */
-#define BROKER_ENDPOINT_LENGTH              ( ( uint16_t ) ( sizeof( BROKER_ENDPOINT ) - 1 ) )
+#define BROKER_ENDPOINT_LENGTH                  ( ( uint16_t ) ( sizeof( BROKER_ENDPOINT ) - 1 ) )
 
 /**
  * @brief Length of client identifier.
  */
-#define CLIENT_IDENTIFIER_LENGTH            ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) - 1 ) )
+#define CLIENT_IDENTIFIER_LENGTH                ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) + MAX_RAND_NUMBER_DIGITS_FOR_CLIENT_ID - 1 ) )
```

Comment:
This doesn't seem entirely accurate anymore
```suggestion
#define MAX_CLIENT_IDENTIFIER_LENGTH            ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) + MAX_RAND_NUMBER_DIGITS_FOR_CLIENT_ID - 1 ) )
```
or even:
```suggestion
#define MAX_CLIENT_IDENTIFIER_SIZE              ( ( uint16_t ) ( sizeof( CLIENT_IDENTIFIER ) + MAX_RAND_NUMBER_DIGITS_FOR_CLIENT_ID ) )
```
and that way, you won't need the `+ 1` in the array below

Reply:
The +1 is added for the NULL character

Reply:
I know, isn't it only used in one place?
```
char clientIdBuffer[ MAX_CLIENT_IDENTIFIER_LENGTH + 1u ];
```
Removing the `- 1` would allow you to remove the `+ 1` from the array.

Reply:
Updated the macro name to `CLIENT_IDENTIFIER_BUFFER_SIZE`